contracts/jobs/JobsManager.sol

```diff
@@ -70,94 +163,300 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
     }
 
     /*
-     * @dev Return transcode claims details for a job
+     * @dev Slash transcoder for missing verification
      * @param _jobId Job identifier
+     * @param _claimId Claim identifier
+     * @param _segmentNumber Segment that was not verified
      */
-    function getJobTranscodeClaimsDetails(uint256 _jobId) public constant returns (uint256, uint256, uint256, uint256, bytes32) {
-        return jobs.getJobTranscodeClaimsDetails(_jobId);
+    function missedVerificationSlash(uint256 _jobId, uint256 _claimId, uint256 _segmentNumber) external returns (bool) {
+        Job storage job = jobs[_jobId];
+        Claim storage claim = job.claims[_claimId];
+
+        // Must be after verification period
+        require(block.number > claim.endVerificationBlock);
+        // Must be before end of slashing period
+        require(block.number <= claim.endSlashingBlock);
+        // Claim must be pending
+        require(claim.status == ClaimStatus.Pending);
+        // Segment must be eligible for verification
+        require(JobLib.shouldVerifySegment(_segmentNumber, claim.segmentRange, claim.claimBlock, verificationRate));
+        // Transcoder must have missed verification for the segment
+        require(!claim.segmentVerifications[_segmentNumber]);
+
+        refundBroadcaster(_jobId, _claimId);
+
+        // Slash transcoder and provide finder params
+        bondingManager().slashTranscoder(job.transcoderAddress, msg.sender, missedVerificationSlashAmount, finderFee);
+
+        // Set claim as slashed
+        claim.status = ClaimStatus.Slashed;
+
+        return true;
     }
 
     /*
-     * @dev End a job. Can be called by either a broadcaster or transcoder of a job
+     * @dev Distribute fees for a particular claim
      * @param _jobId Job identifier
+     * @param _claimId Claim identifier
      */
-    function endJob(uint256 _jobId) external returns (bool) {
-        return jobs.endJob(_jobId, jobEndingPeriod);
+    function distributeFees(uint256 _jobId, uint256 _claimId) public returns (bool) {
+        // Must be valid job id
+        require(_jobId < numJobs);
+        // Must be valid claim id
+        require(_claimId < jobs[_jobId].claims.length);
+
+        Job storage job = jobs[_jobId];
+        Claim storage claim = job.claims[_claimId];
+
+        // Sender must be elected transcoder for job
+        require(job.transcoderAddress == msg.sender);
+        // Claim must not be complete
+        require(claim.status == ClaimStatus.Pending);
+        // Slashing period must be over for claim
+        require(claim.endSlashingBlock < block.number);
+
+        uint256 fees = claim.segmentRange[1].sub(claim.segmentRange[0]).add(1).mul(job.maxPricePerSegment);
+        // Deduct fees from escrow
+        job.escrow = job.escrow.sub(fees);
+        // Add fees to transcoder's fee pool
+        bondingManager().updateTranscoderFeePool(msg.sender, fees, claim.claimBlock, claim.transcoderTotalStake);
```

Comment:
Do we need to assign the transcoder their cut of the fees here, or is that taken care of somewhere else?